xbmc/addons/interfaces/GUI/Window.cpp

```diff
@@ -278,6 +281,63 @@ bool Interface_GUIWindow::do_modal(void* kodiBase, void* handle)
   return true;
 }
 
+void Interface_GUIWindow::clear_item_list(void* kodiBase, void* handle)
+{
+  CAddonDll* addon = static_cast<CAddonDll*>(kodiBase);
+  CGUIAddonWindow* pAddonWindow = static_cast<CGUIAddonWindow*>(handle);
+  if (!addon || !pAddonWindow)
+  {
+    CLog::Log(LOGERROR, "Interface_GUIWindow::%s - invalid handler data (kodiBase='%p', handle='%p') on addon '%s'",
+                          __FUNCTION__, addon, handle, addon ? addon->ID().c_str() : "unknown");
+    return;
+  }
+
+  Interface_GUIGeneral::lock();
+  static_cast<CGUIAddonWindow*>(handle)->ClearList();
```

Comment:
You should use variable 'pAddonWindow' here as it already contains the value of <code>static_cast<CGUIAddonWindow*>(handle)</code>.

Same glitch is present in other functions in this source file.